core/drivers/stm32_i2c.c

```diff
@@ -1383,6 +1383,10 @@ static int i2c_read(struct i2c_handle_s *hi2c, struct i2c_request *request,
 	if (i2c_wait_stop(hi2c, timeout_ref))
 		goto bail;
 
+	/* Clear the NACK generated at the end of the transfer */
+	if ((io_read32(get_base(hi2c) + I2C_ISR) & I2C_ISR_NACKF))
+		io_write32(get_base(hi2c) + I2C_ICR, I2C_ICR_NACKCF);
```

Comment:
This looks good to me but I've asked a 2nd eye on this. Please stay tune, I'll answer asap.

Reply:
thanks @etienne-lms

Reply:
perhaps we should just clear it unconditionally

Reply:
@etienne-lms just a reminder that this fix is pending - ie, the current driver upstream is broken in this xfer mode.